### DISCUSSION TITLE
How to make a light gray text conversion to Group4 readable?

### DISCUSSION THREAD

**Discussion by valdier1** [ImageMagick/ImageMagick#5204] [Help] [answered]
I'm having trouble getting decent image quality when converting a 3 color gray image to Group4 tif. I've tried using level and not success, but I know there must be a way to make this readable. I'm attaching the included file if anyone has experience with fixing these levels?

magick mogrify -compress Group4 img_8807a.tif

[IMG_8807a.zip](https://github.com/ImageMagick/ImageMagick/files/8848426/IMG_8807a.zip)

**Comment by snibgo** [comment]
I suggest you start by making the ink black, and the paper white.
```
magick IMG_8807a.jpg -contrast-stretch 2,80% -compress Group4 x2.tiff
```

**Comment by fmw42** [comment, marked as answer]
try

`magick IMG_8807a.jpg -negate -lat 20x20+10% -negate -monochrome -compress Group4 result.tiff`

**Reply by valdier1** [reply]
Wow... this result is fantastic... I seriously need to read and understand what exactly you did here

**Reply by fmw42** [reply]
-lat is an adaptive thresholding or local thresholding technique. It has to have white letters on black so I invert using -negate before and after. The 20x20 is the region size which should be larger than the letters (font size). The 10% is the threshold level.  Adapt as needed.  The -monochrome is the method of turning gray to black.

**Reply by Jmuccigr** [reply]
Curious about the details here. In particular how much work is the `-monochrome` doing? That is, the compression to Group4 will make the image monochrome and an application `-threshold` will (I think) help determine where the line lies between black and white, so is the `-monochrome` strictly necessary?

**Reply by fmw42** [reply]
It was more useful when it did dithering. Right now the dithering has been lost and it is just doing some kind of effective thresholding, though likely is not that simple.
